File: lab8/bubblesort.py
def main():
   l= [10,2,4,5,2,3,6,90]
   print(l)
   print(bubblesort(l))

def bubblesort(array):
   n = len(array)
   swapped = True
   temp = 0
   while (swapped == True):
      swapped = False
      for i in range(1,n-1,1):
         if array[i-1]>array[i]:
            temp = array[i-1]
            array[i-1]=array[i]
            array[i]= temp
            swapped = True
   status = True
   return [status, array]

main()

File: lab8/bubblesort.c
#include<stdio.h>

int bubblesort(int *array, int size);

int main(void){
   int i;
   int m[10]={3,2,5,33,6,4,2,7,8,12};
   for (i = 0; i<10; i++){
      printf("%d ",m[i]);
   }
   printf("\n");
   bubblesort(m, 10);
   for (i=0; i<10; i++){
      printf("%d ",m[i]);
   }
}

int bubblesort(int *array, int size){
   int n = size;
   int swapped = 1;
   int i; 
   int temp;
   if (array == NULL){
      return -1;
   }
   while (swapped == 1){
      swapped = 0;
      for (i=1; i<n; i++){
         if (array[i-1]>array[i]){
            temp = array[i-1];
            array[i-1] = array[i];
            array[i]=temp;
            swapped = 1;
         }
      }
   }
   return 1;
}


